openstack/ces/v1/metricdata/results.go

```diff
@@ -4,97 +4,112 @@ import (
 	"github.com/opentelekomcloud/gophertelekomcloud"
 )
 
-type MetricData struct {
-	// Specifies the namespace in service.
+type BatchMetricData struct {
+	// Specifies the metric namespace.
+	// The value must be in the service.item format and can contain 3 to 32 characters.
 	Namespace string `json:"namespace"`
-
-	// The value can be a string of 1 to 64 characters
-	// and must start with a letter and contain only uppercase
-	// letters, lowercase letters, digits, and underscores.
+	// Specifies the metric name. Start with a letter. Enter 1 to 64 characters.
 	MetricName string `json:"metric_name"`
-
-	// Specifies the list of the metric dimensions.
-	Dimensions []Dimension `json:"dimensions"`
-	Datapoints []Data      `json:"datapoints"`
-
+	// Specifies the list of metric dimensions.
+	Dimensions []MetricsDimension `json:"dimensions"`
+	// Specifies the metric data list.
+	// Since Cloud Eye rounds up the value of from based on the level of granularity for data query,
+	// datapoints may contain more data points than expected.
+	Datapoints []DatapointForBatchMetric `json:"datapoints"`
 	// Specifies the metric unit.
 	Unit string `json:"unit"`
 }
 
-type Dimension struct {
-	Name  string `json:"name"`
-	Value string `json:"value"`
+type DatapointForBatchMetric struct {
+	// Specifies the maximum value of metric data within a rollup period.
+	Max float64 `json:"max,omitempty"`
+	// Specifies the minimum value of metric data within a rollup period.
+	Min float64 `json:"min,omitempty"`
+	// Specifies the average value of metric data within a rollup period.
+	Average float64 `json:"average,omitempty"`
+	// Specifies the sum of metric data within a rollup period.
+	Sum float64 `json:"sum,omitempty"`
+	// Specifies the variance of metric data within a rollup period.
+	Variance float64 `json:"variance,omitempty"`
+	// Specifies when the metric is collected. It is a UNIX timestamp in milliseconds.
+	Timestamp int64 `json:"timestamp"`
```

Comment:
Should be `int`, as discussed

Reply:
The timestamp must be `long`, no matter how TF does.

Reply:
Ok, that's good, haven't payed attention it's a timestamp